src/main/java/microsoft/exchange/webservices/data/core/request/HangingServiceRequestBase.java

```diff
@@ -239,13 +238,26 @@ private void parseResponses() {
       //simply results in a long-running connection.
       this.disconnect(HangingRequestDisconnectReason.UserInitiated, null);
     } catch (Exception ex) {
-      // Stream is closed, so disconnect.
-      this.disconnect(HangingRequestDisconnectReason.Exception, ex);
+      if (!(disconnecting && isSocketClosed(ex))) {
+        // Stream was closed due to an error
+        this.disconnect(HangingRequestDisconnectReason.Exception, ex);
+      }
+
     } finally {
       IOUtils.closeQuietly(responseCopy);
     }
   }
 
+  private static boolean isSocketClosed(Throwable e) {
```

Comment:
Is this about getting smth like the [rootcause](https://commons.apache.org/proper/commons-lang/javadocs/api-2.4/org/apache/commons/lang/exception/ExceptionUtils.html#getRootCause(java.lang.Throwable))?